river/command/rule.zig

```diff
@@ -123,41 +143,51 @@ pub fn listRules(_: *Seat, args: []const [:0]const u8, out: *?[]const u8) Error!
+        .float => server.config.float_rules.getMaxGlobLen(),
+        .ssd => server.config.ssd_rules.getMaxGlobLen(),
+        .tag => server.config.tag_rules.getMaxGlobLen(),
     };
-
-    var action_column_max = "action".len;
-    var app_id_column_max = "app-id".len;
-    for (rules) |rule| {
-        const action = switch (list) {
-            .float => if (rule.value) "float" else "no-float",
-            .ssd => if (rule.value) "ssd" else "csd",
-        };
-        action_column_max = @max(action_column_max, action.len);
-        app_id_column_max = @max(app_id_column_max, rule.app_id_glob.len);
-    }
-    action_column_max += 2;
-    app_id_column_max += 2;
+    const app_id_column_max = 2 + @max("app-id".len, max_glob_len.app_id);
+    const title_column_max = 2 + @max("title".len, max_glob_len.title);
 
     var buffer = std.ArrayList(u8).init(util.gpa);
     const writer = buffer.writer();
 
-    try fmt.formatBuf("action", .{ .width = action_column_max, .alignment = .Left }, writer);
+    try fmt.formatBuf("title", .{ .width = title_column_max, .alignment = .Left }, writer);
     try fmt.formatBuf("app-id", .{ .width = app_id_column_max, .alignment = .Left }, writer);
-    try writer.writeAll("title\n");
-
-    for (rules) |rule| {
-        const action = switch (list) {
-            .float => if (rule.value) "float" else "no-float",
-            .ssd => if (rule.value) "ssd" else "csd",
-        };
-        try fmt.formatBuf(action, .{ .width = action_column_max, .alignment = .Left }, writer);
-        try fmt.formatBuf(rule.app_id_glob, .{ .width = app_id_column_max, .alignment = .Left }, writer);
-        try writer.print("{s}\n", .{rule.title_glob});
+    try writer.writeAll("action\n");
+
+    switch (list) {
+        .float, .ssd => {
+            const rules = switch (list) {
+                .float => server.config.float_rules.rules.items,
+                .ssd => server.config.ssd_rules.rules.items,
+                else => unreachable,
+            };
+            for (rules) |rule| {
+                try fmt.formatBuf(rule.title_glob, .{ .width = title_column_max, .alignment = .Left }, writer);
+                try fmt.formatBuf(rule.app_id_glob, .{ .width = app_id_column_max, .alignment = .Left }, writer);
+                try writer.print("{s}\n", .{switch (list) {
+                    .float => if (rule.value) "float" else "no-float",
+                    .ssd => if (rule.value) "ssd" else "csd",
+                    else => unreachable,
+                }});
+            }
+        },
+        .tag => {
+            const rules = server.config.tag_rules.rules.items;
+            for (rules) |rule| {
+                try fmt.formatBuf(rule.title_glob, .{ .width = title_column_max, .alignment = .Left }, writer);
+                try fmt.formatBuf(rule.app_id_glob, .{ .width = app_id_column_max, .alignment = .Left }, writer);
+                try writer.print("{}\n", .{rule.value});
```

Comment:
Would it be nicer to print this in binary with `{b}`?

It would admittedly be prettier to right-align the numbers in the tag column though that would require determining their max width when printed. If printing in binary the printed width is `std.math.log2(value)`.

Reply:
Not sure if binary would be better. Right now all other commands use the integer representation, so for consistency I think it makes sense here as well.

Reply:
I think the base 10 representation makes sense for taking a tag value as input as it allows users to use the standard bitshifting techniques to construct a set of tags.

I really don't think viewing a set of tags in base 10 is that useful though. It seems to me that a user seeing a `5` in the output of this list could be quite confused when the view appears on the tags commonly 1 and 3.

Instead displaying the base 2 representation of `101` seems less likely to be confusing IMO. I suppose there's still the chance that they might expect it to appear on tag one hundred and one, we could consider adding leading zeros to offset that but I'm not convinced that would be worthwhile.